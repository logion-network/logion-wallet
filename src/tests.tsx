import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };

export async function clickByName(name: string | RegExp | ((accessibleName: string, element: Element) => boolean)) {
    let button: HTMLElement;
    await waitFor(() => button = screen.getByRole("button", { name }));
    await userEvent.click(button!);
}

export async function typeByLabel(label: string, value: string) {
    let input: HTMLElement;
    await waitFor(() => input = screen.getByLabelText(label));
    await userEvent.type(input!, value);
}
