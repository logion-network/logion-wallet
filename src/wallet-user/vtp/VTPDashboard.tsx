import { FullWidthPane } from "../../common/Dashboard";
import { useCommonContext } from "../../common/CommonContext";
import IconTextRow from "../../common/IconTextRow";
import Icon from "../../common/Icon";
import Frame from "../../common/Frame";
import Tabs from "../../common/Tabs";
import { useState } from "react";
import VTPOpenedLocs from "./VTPOpenedLocs";
import VTPClosedLocs from "./VTPClosedLocs";

export default function VTPDashboard() {
    const { colorTheme } = useCommonContext();
    const [ locTabKey, setLocTabKey ] = useState<string>('open');
    return (
        <FullWidthPane
            className="VTPDashboard"
            mainTitle={ "Third Party LOC Management" }
            titleIcon={ {
                icon: {
                    id: 'vtp-icon'
                },
                background: colorTheme.topMenuItems.iconGradient,
            } }
        >
            <IconTextRow
                icon={ <Icon icon={ { id: 'tip' } } width="45px" /> }
                text={
                    <p>
                        You have been nominated as <strong>“Verified Third Party”</strong> by at least one Logion Legal
                        Officer.<br />
                        This means that you have the responsibility to contribute either public data and/or confidential
                        documents in Legal Officer Cases (LOC) managed by related Logion Legal Officers.<br />
                        You understand that the material you provide in such context shall
                        be <strong>authentic</strong> and <strong>accurate</strong>.<br />
                        With regard to future public data, you confirm that the submitted public data can be publicly
                        exposed, especially under GDPR regulation or any kind of data protection regulation that may
                        apply. Any kind of personal (eg: name, email, phone number, etc.) and/or confidential
                        information shall be strictly excluded from public data under your responsibility. Files will
                        not be public (their hash will).</p>

                } />
            <Frame title="Third party LOC list">
                <Tabs
                    activeKey={ locTabKey }
                    onSelect={ key => setLocTabKey(key || 'open') }
                    tabs={ [
                        {
                            key: "open",
                            title: "Open",
                            render: () => <VTPOpenedLocs/>
                        },
                        {
                            key: "closed",
                            title: "History",
                            render: () => <VTPClosedLocs/>
                        },
                    ] }
                />
            </Frame>
        </FullWidthPane>
    );
}
