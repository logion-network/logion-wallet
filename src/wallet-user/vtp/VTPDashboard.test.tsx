import { shallowRender } from "../../tests";
import VTPDashboard from "./VTPDashboard";

jest.mock('../../common/CommonContext');

describe("VTPDashboard", () =>  {

    it("renders", () => {
        const tree = shallowRender(<VTPDashboard/>);
        expect(tree).toMatchSnapshot();
    })
})
