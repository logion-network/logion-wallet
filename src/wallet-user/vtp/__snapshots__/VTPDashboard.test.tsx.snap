// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTPDashboard renders 1`] = `
<FullWidthPane
  className="VTPDashboard"
  mainTitle="Third Party LOC Management"
  titleIcon={
    Object {
      "background": Object {
        "from": "#3b6cf4",
        "to": "#6050dc",
      },
      "icon": Object {
        "id": "vtp-icon",
      },
    }
  }
>
  <IconTextRow
    icon={
      <Icon
        icon={
          Object {
            "id": "tip",
          }
        }
        width="45px"
      />
    }
    text={
      <p>
        You have been nominated as 
        <strong>
          “Verified Third Party”
        </strong>
         by at least one Logion Legal Officer.
        <br />
        This means that you have the responsibility to contribute either public data and/or confidential documents in Legal Officer Cases (LOC) managed by related Logion Legal Officers.
        <br />
        You understand that the material you provide in such context shall be 
        <strong>
          authentic
        </strong>
         and 
        <strong>
          accurate
        </strong>
        .
        <br />
        With regard to future public data, you confirm that the submitted public data can be publicly exposed, especially under GDPR regulation or any kind of data protection regulation that may apply. Any kind of personal (eg: name, email, phone number, etc.) and/or confidential information shall be strictly excluded from public data under your responsibility. Files will not be public (their hash will).
      </p>
    }
  />
  <Frame
    title="Third party LOC list"
  >
    <Tabs
      activeKey="open"
      onSelect={[Function]}
      tabs={
        Array [
          Object {
            "key": "open",
            "render": [Function],
            "title": "Open",
          },
          Object {
            "key": "closed",
            "render": [Function],
            "title": "History",
          },
        ]
      }
    />
  </Frame>
</FullWidthPane>
`;
